joomla/joomla-cms
# [5.0] Smart Search: Index custom fields

Pull Request for Issue #12829. Re-do of #36747.

### Summary of Changes
This PR implements indexing of the custom fields for a component. This is controlled on a per field basis, where you can select if the content should be just added to the search index or handled as a taxonomy or both or actually none of the above. You can then implement this feature for your respective finder plugin with a single line of code.
Each custom field has a new option in its edit screen where you can select how the indexer should treat this field. The finder plugin then has to call `Helper::addCustomFields($item, 'context');` and the rest is just magic. ;)
Right now this is implemented for com_content and com_contacts.

This is a re-do of #36747, which had to be reverted due to issues when indexing via CLI. This issue should be solved when #38585 and #39550 are merged.

### Testing Instructions
0. Make sure that #38585 and #39550 are merged.
1. Your installation should have some custom fields for com_content or com_contact in it.
2. Edit a custom field and set the "Smart Search" option to "Make searchable and add as taxonomy".
3. Run the indexing process either by running the mass indexer in the smart search component or by editing an item and saving it again.
4. See that the custom field shows up as a taxonomy and can be find by searching for its value.

### Documentation Changes Required
The documentation needs to be amended to contain how to implement this feature for third party finder plugins.
